src/pydicom/pixel_data_handlers/util.py

```diff
@@ -1016,27 +1018,29 @@ def get_expected_length(ds: "Dataset", unit: str = "bytes") -> int:
     columns = cast(int, ds.Columns)
     samples_per_pixel = cast(int, ds.SamplesPerPixel)
     bits_allocated = cast(int, ds.BitsAllocated)
+    pixel_data_buffer_length = cast(int, len(ds.PixelData))
```

Comment:
Reliance on the length of *Pixel Data* will return incorrect results for compressed transfer syntaxes.